R/morphemepiece.R

```diff
@@ -196,38 +305,243 @@ magrittr::`%>%`
 #' Tokenize Sequence with Morpheme Pieces
 #'
 #' Given a single sequence of text and a morphemepiece vocabulary, tokenizes the
-#' text. (This is *not* fully functional yet. It's mainly a placeholder so that
-#' the package exports something.)
-#' 
+#' text.  
 #'
-#' @inheritParams .mp_tokenize_word
+#' @inheritParams .mp_tokenize_word_lookup
 #' @param text Character scalar; text to tokenize.
 #'
 #' @return A character vector of tokenized text (later, this should be a named
 #' integer vector, as in the wordpiece package.)
 #' @export
 morphemepiece_tokenize <- function(text,
-                                   vocab) {
-
+                                   vocab,
+                                   lookup,
+                                   unk_token = "[UNK]",
+                                   max_chars = 100) {
+  is_cased <- attr(vocab, "is_cased")
+  if (!is_cased) {
     text <- tolower(text)
+  }
+  
+  text <- .convert_to_unicode(text)
+  text <- .clean_text(text)
+  text <- .tokenize_chinese_chars(text)
+  text <- .strip_accents(text)
+  text <- .split_on_punc(text)
+  text <- purrr::map(wordpiece:::.whitespace_tokenize(text),
+                     .f = .mp_tokenize_word_lookup, 
+                     vocab = vocab, 
+                     lookup = lookup)
+  text <- unlist(text)
+  return(text)
+  #  For testing on datascience:
```

Comment:
Should we clean these out?

Reply:
eventually, yeah. I left those in partly so that you could quickly try it out if you wanted. :)